src/display.hpp

```diff
@@ -583,9 +583,62 @@ class display : public filter_context, public video2::draw_layering
 	map_labels& labels();
 	const map_labels& labels() const;
 
+	/** Holds options for calls to function 'announce' (@ref announce). */
+	class announce_options {
+	private:
+		/** Lifetime measured in frames. Field. */
+		int lifetime_;
+
+		/**
+		 * An announcement according these options should replace the
+		 * previous announce (typical of fast announcing) or not
+		 * (typical of movement feedback). Field.
+		 */
+		bool discard_previous_;
+
+	public:
+		/** Enabled default constructor. */
+		announce_options() {
+			lifetime_ = 100;
+			discard_previous_ = true;
+		}
+
+		/** Lifetime measured in frames. Field setter. */
+		announce_options lifetime(const int lifetime) {
+			lifetime_ = lifetime;
+			return * this;
+		}
+
+		/**
+		 * An announcement according these options should replace the
+		 * previous announce (typical of fast announcing) or not
+		 * (typical of movement feedback). Field setter.
+		 */
+		announce_options discard_previous(
+				const bool discard_previous) {
```

Comment:
No need for a line break here. Also, please put the leading function brace on its own line.